### ISSUE TITLE
Getting a black page on Chrome but not Firefox and Safari

### ISSUE THREAD

**Issue by Timothee** [codegl/tracegl#107]
On Chrome Version 31.0.1612.2 dev for Mac, I'm getting a fully black screen (if not for a single salmon-colored pixel in the top left) when I try to look at the traces.

In the devtools console, I get the following error:

``` javascript
Uncaught Error: Could not link, max varying:15
precision mediump float;
#define ucol vec4
uniform vec4 ps;
uniform vec4 sz;
uniform float fg;
uniform sampler2D T;
vec2 f;
uniform vec2 s;
uniform ucol N_g;
varying vec4 rv_;
vec4 r;
vec4 _f(){
vec3 v = vec3(floor(ps.z+(ps.x+r.x)*sz.x),floor(ps.w+(ps.y+r.y)*sz.y-sz.z),ps.z+(ps.x+r.z)*sz.x);
vec4 c = vec4((texture2D(T,vec2((fg),0.))));
;
if(f.x<v.x+0.5*sz.x){
vec2 d = vec2(f.x-(v.x+sz.x),f.y-(v.y+0.5*sz.y));
;
if(d.y<0.&&mod(r.w,2.)==1.)return vec4(c);
;
if(d.y>0.&&mod(r.w,4.)>=2.)return vec4(c);
return vec4(c.xyz,sqrt(d.x*d.x+d.y*d.y)>9.?0.:c.w);
;
}
else ;
if(f.x>v.z-0.5*sz.x){
vec2 d = vec2(f.x-(v.z-sz.x),f.y-(v.y+0.5*sz.y));
;
if(d.y<0.&&mod(r.w,8.)>=4.)return vec4(c);
;
if(d.y>0.&&mod(r.w,16.)>=8.)return vec4(c);
return vec4(c.xyz,sqrt(d.x*d.x+d.y*d.y)>9.?0.:c.w);
;
}
return vec4(c);
;
}
void main(void){
 r = rv_;
f = vec2(gl_FragCoord.x,s.y-gl_FragCoord.y);
 gl_FragColor = _f();
}
 (index):3973
frag (index):3973
gl.getShader (index):3934
listView (index):10605
(anonymous function) (index):12328
p.set (index):5873
ct.hSplit (index):8446
(anonymous function) (index):12327
p.set (index):5873
group (index):5391
(anonymous function) (index):12322
p.set (index):5873
ct.vSplit (index):8476
(anonymous function) (index):12319
p.set (index):5873
group (index):5391
(anonymous function) (index):12257
gl.load (index):1199
(anonymous function) (index):11994
(anonymous function) (index):12461

```

Safari and Firefox work fine however.

Anybody else is seeing this?

**Comment by figmentfel** [comment]
@Timothee: I have the exact same result as you ... black screen, single salmon-colored pixel (top-left) and got the same error in devconsole -> line 3973

EDIT: I'm on Chrome Version 31.0.1624.0 canary

**Comment by IngageStroliaC** [comment]
Pretty much what I'm getting too. I also commented here:https://github.com/codegl/tracegl/issues/95#issuecomment-25017646

**Comment by LiamKarlMitchell** [comment]
This has been unresolved for 25+ days?
Awh man...
I am using Chrome Version 30.0.1599.69 
Any fix/suggestions?

Edit: 
Line 5976 Make the following changes.

if (gl.getProgramParameter(sp, gl.LINK_STATUS)) {
  gl.useProgram(sp)
}
else
{
  console.warn("Could not link, max varying:" + gl.getParameter(gl.MAX_VARYING_VECTORS) +"\n"+ gl.getShaderInfoLog(gf) + fs);
}

Now to actually try this thing out on my project :)

**Comment by Timothee** [comment]
@LiamKarlMitchell nice I'll have to try that. I never got around to try to fix it, but it'd be great if that works. Thanks!

**Comment by Timothee** [comment]
@LiamKarlMitchell it works! :confetti_ball: Many thanks! Of course, I get a bunch of errors `Could not link…` and some consequent errors since the program couldn't be linked but that's better than nothing.

**Comment by LiamKarlMitchell** [comment]
Yeah :) However in my version of chrome I do not get the colors such as Green or Red as shown in video.
Yet on another chrome I tried it worked. Newer versions seem buggy :( bleh. Firefox seems to work perfectly.
Try version 27 of chrome. Or maybe theres some sort of experimental flag that I have to set.

**Comment by djmuri** [comment]
Using latest Firefox and Windows 7, node tracegl helloworld.js requested as http://localhost:2000 returns 100% black screen with no errors, what am I doing incorrectly? My CMD window prints every 2 seconds "Hello World" on the screen as expected. Window 7 is German

**Comment by djmuri** [comment]
ok,  got something on Win8 with Firefox 24, have four panes but can not read output , letters appears as commas.

**Comment by LiamKarlMitchell** [comment]
Using firefox that I downloaded the other night from their site.
Version 2.4
Im on Win7

See image.
http://i.imgur.com/Q0j8VG3.png
